src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -250,7 +285,7 @@ private bool TryExtractStringConcatArgs(BoundExpression lowered, out ImmutableAr
 
         private static bool IsNullOrEmptyStringConstant(BoundExpression operand)
         {
-            return (operand.ConstantValueOpt != null && string.IsNullOrEmpty(operand.ConstantValueOpt.StringValue)) ||
+            return (operand.ConstantValueOpt?.IsString == true && string.IsNullOrEmpty(operand.ConstantValueOpt.StringValue)) ||
                     operand.IsDefaultValue();
```

Comment:
>operand.IsDefaultValue() [](http://example.com/codeflow?start=20&length=24)

I am not sure what motivated the change in the first sub-expression, but if the type becomes important, then, perhaps, we should check it in this part too.

Reply:
I suspect this subexpression is going to return true for ```'\0'```. And then the behavior of ```TryFoldTwoConcatOperands``` will be broken, quite possibly in several different ways.